## Feature and labels musical example

* Features: intensity, tempo, genre, gender of the voice
* Labels: like, don't like

## Cancer test
$P(c)$ = 0.01
* 90% it's **positive** if you have C (*sensitivity*);
* 90% it's **negative** if you don't have C (*specitivity*)
* test = positve
* Probability of having cancer?

## Bayes rule
Prior probability * test evidence = posterior 

**prior**:
* $P(C)$=0.01; $P(¬C)$=0.99;
* $P(+|C)$=0.9; $P(-|C)$=0.1;
* $P(-|¬C)$=0.9; $P(+|¬C)$=0.1;

**joint posterior**:

$P(C,+)=P(C)*P(+|C)$ this is the branch for the consideration that you have cancer (cancer hypothesis)

$P(¬C,+)=P(¬C)*P(+|¬C)$ this is the branch for the consideration that you have no cancer (no cancer hypothesis)

**normalizes**:

$P(+)=P(C,+)+P(¬C,+)$ the total probability that the test is positive

**extra posterior**:

$P(C|+)=P(C,+)/P(+)$

$P(¬C|Pos)=P(¬C,+)/P(+)$

In [2]:
c=0.01
n_c=0.99
pos_c=0.9
neg_c=0.1
neg_n_c=0.9
pos_n_c=0.1
c_pos=c*pos_c
n_c_pos=n_c*pos_n_c
c_pos, n_c_pos

(0.009000000000000001, 0.099)

In [4]:
pos = c_pos+n_c_pos
pos

0.10800000000000001

In [8]:
p_c_pos = c_pos/pos

In [9]:
p_n_c_pos = n_c_pos/pos

In [10]:
p_c_pos + p_n_c_pos

1.0

## Bayes rule for classification
Text learning
* P(Chris) = 0.5
> love=0.1; deal=0.8; life=0.1
* P(Sara) = 0.5
> love=0.3; deal=0.2; life=0.3

love life

life deal

In [15]:
chris,sara=[0.5, 0.5]
love_c,deal_c,life_c = [0.1, 0.8, 0.1]
love_s,deal_s,life_s = [0.5, 0.2, 0.3]

In [16]:
# love life
c1 = love_c*life_c*chris
s1 = love_s*life_s*sara
c1, s1

(0.005000000000000001, 0.075)

In [17]:
# life deal
c2 = life_c*deal_c*chris
s2 = life_s*deal_s*sara
c2, s2

(0.04000000000000001, 0.03)

In [20]:
# Posterior probability of Chris and Sara for "life deal"
c_ld = c2/(c2+s2)
s_ld = s2/(c2+s2)
c_ld, s_ld

(0.5714285714285715, 0.4285714285714285)

In [21]:
# love deal
c3 = love_c*deal_c*chris
s3 = love_s*deal_s*sara
c_lvd = c3/(c3+s3)
s_lvd = s3/(c3+s3)
c_lvd, s_lvd

(0.4444444444444445, 0.5555555555555555)

## Why is naive bayes naive?
Let's you identify from text source wheter a label is more likely than other, but it ignores words order (just the frequencies).

Result from script:

> no. of Chris training emails: 7936

> no. of Sara training emails: 7884

> ('tempo de treinamento:', 0.957, 's')

> ('tempo de predicao:', 0.108, 's')

> 0.9732650739476678
